python/ray/serve/api.py

```diff
@@ -1800,7 +1800,16 @@ def run(
     ]
 
     client.deploy_group(parameter_group, _blocking=True)
-    return deployments[-1].get_handle()
+
+    serve_handle = []
+    for deployment in deployments:
+        if deployment.route_prefix == "/":
+            serve_handle.append(deployment.get_handle())
+    assert len(serve_handle) == 1, (
+        "Serve Application should have one and only one deployment with "
```

Comment:
This message seems to imply an Application can have multiple deployments with non-`/` route-prefixes. Can we say that only one deployment can have a non-`None` prefix?

Reply:
also, this might be a user-facing error so please don't use `assert`